astropy/io/fits/tests/test_image.py

```diff
@@ -925,9 +926,14 @@ def test_image_update_header(self):
         with fits.open(self.temp("test0.fits")) as hdul:
             assert (orig_data == hdul[1].data).all()
 
-    # The test below raised a `ResourceWarning: unclosed transport` exception
-    # due to a bug in PYTHON_LT_3_11 (cf. cpython#90476)
-    @pytest.mark.filterwarnings("ignore:unclosed transport <asyncio.sslproto")
+    if sys.version_info < (3, 11):
+        # The test below raised a `ResourceWarning: unclosed transport` exception
+        # due to a bug in Python < 3.11 (cf. cpython#90476)
+        mark = pytest.mark.filterwarnings("ignore:unclosed transport <asyncio.sslproto")
+    else:
+        mark = lambda f: f
+
+    @mark
```

Comment:
Instead of introducing no-op marker, maybe cleaner to use `warnings.filterwarnings` context manager inside this function?

Reply:
It would, only I don't which part of this long test is supposed to trigger that warning (and I can't seem to reproduce it locally), so this is seemed to me that a no-op decorator would at least keep the diff minimal.

Reply:
I was gonna suggest the whole test go inside the context manager but yeah, would be larger diff. Leave it as is then. Thanks!